### PR TITLE
[iam] Drop OIDC prefix from service, remove unused handler

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Touch ups on the OIDC service. See inline comments for reasoning.

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/iam/pkg/oidc/service.go [hunk 1]
```diff
@@ -15,59 +15,56 @@ import (
...
 	"github.com/gitpod-io/gitpod/common-go/log"
-	"github.com/go-chi/chi/v5"
 	"golang.org/x/oauth2"
 )
 
-type OIDCService struct {
-	Handler chi.Router
-
-	configsById      map[string]*OIDCClientConfig
+type Service struct {
```
Comment (easyCZ) [inline components/iam/pkg/oidc/service.go:21]: The struct is used as `oidc.Service`, there's no need to duplicate the OIDC in the name again. This change aligns it with go standard guidance.

Comment (easyCZ) [reply components/iam/pkg/oidc/service.go:21]: And similarly for the objects below.

File: components/iam/pkg/oidc/service.go [hunk 2]
```diff
@@ -15,59 +15,56 @@ import (
...
-	s.configsById = make(map[string]*OIDCClientConfig)
-	s.verifierByIssuer = make(map[string]*oidc.IDTokenVerifier)
-	s.providerByIssuer = make(map[string]*oidc.Provider)
-	return &s
+func NewService() *Service {
+	return &Service{
+		configsById:      map[string]*ClientConfig{},
+		verifierByIssuer: map[string]*oidc.IDTokenVerifier{},
+		providerByIssuer: map[string]*oidc.Provider{},
+	}
```
Comment (easyCZ) [inline components/iam/pkg/oidc/service.go:49]: For cases where we need to initialize maps, this is the more common pattern. In general, direct field assignment is used sparingly as it easily leads to missing a field which gets added in the future.

File: components/iam/pkg/server/server.go [hunk 3]
```diff
@@ -35,7 +35,7 @@ func Start(logger *logrus.Entry, version string, cfg *config.ServiceConfig) erro
 	}
 
 	// Requests to /oidc/* are handled by oidc.Router
-	oidcService := oidc.NewOIDCService()
+	oidcService := oidc.NewService()
```
Comment (easyCZ) [inline components/iam/pkg/server/server.go:38]: This is an example where the package already provides the `OIDC` context, and hence not needed in the struct name.

**Review by AlexTugarev — APPROVED** [review]

**Pull request merged**